external-crates/move/crates/move-vm-types/src/values/values_impl.rs

```diff
@@ -1205,10 +1205,22 @@ impl SignerRef {
  *
  **************************************************************************************/
 impl Locals {
-    pub fn new(n: usize) -> Self {
-        Self(Rc::new(RefCell::new(
-            iter::repeat_with(|| ValueImpl::Invalid).take(n).collect(),
-        )))
+    // Legacy -- will be deprecated.
+    pub fn new(size: usize) -> Self {
+        Locals::new_from(vec![], size)
+    }
+
+    /// Creates a new Locals set from a set of parameter values and a total size.
+    /// - `params` is initial params, **in order** (ordered `0, 1, ..., n`).
+    /// - `size` is total size, which should include `params` size.
+    pub fn new_from(params: Vec<Value>, size: usize) -> Self {
+        let invalids_len = size - params.len();
+        let local_values = params
+            .into_iter()
+            .map(|v| v.0)
+            .chain(iter::repeat_with(|| ValueImpl::Invalid).take(invalids_len))
```

Comment:
Maybe 
`(0..invalids_len).map(|_| ValueImpl::Invalid)`

The only reason I would prefer it (or care really) is that the length is up front, rather than at the end